tests/MongoDB.Driver.Core.TestHelpers/CoreExceptionHelper.cs

```diff
@@ -49,30 +53,40 @@ public static Exception CreateException(Type exceptionType)
                         var connectionId = new ConnectionId(serverId, 1);
                         var cursorId = 1L;
                         var query = new BsonDocument();
-                        return new MongoCursorNotFoundException(connectionId, cursorId, query);
+                        exception = new MongoCursorNotFoundException(connectionId, cursorId, query);
                     }
+                    break;
 
                 case "MongoNodeIsRecoveringException":
                     {
                         var clusterId = new ClusterId(1);
                         var serverId = new ServerId(clusterId, new DnsEndPoint("localhost", 27017));
                         var connectionId = new ConnectionId(serverId, 1);
-                        var result = new BsonDocument();
-                        return new MongoNodeIsRecoveringException(connectionId, null, result);
+                        var result = BsonDocument.Parse("{ code : 11600 }"); // InterruptedAtShutdown;
+                        exception = new MongoNodeIsRecoveringException(connectionId, null, result);
                     }
+                    break;
 
                 case "MongoNotPrimaryException":
                     {
                         var clusterId = new ClusterId(1);
                         var serverId = new ServerId(clusterId, new DnsEndPoint("localhost", 27017));
                         var connectionId = new ConnectionId(serverId, 1);
-                        var result = new BsonDocument();
-                        return new MongoNotPrimaryException(connectionId, null, result);
+                        var result = BsonDocument.Parse("{ code : 10107 }"); // NotMaster;
+                        exception = new MongoNotPrimaryException(connectionId, null, result);
                     }
+                    break;
 
                 default:
                     throw new ArgumentException($"Unexpected exception type: {exceptionType.Name}.", nameof(exceptionType));
             }
+
+            if (exception is MongoException mongoException)
+            {
+                RetryabilityHelper.AddRetryableWriteErrorLabelIfRequired(mongoException);
```

Comment:
I will handle this in the scope of CSHARP-3001. Also, it's possible to just temove these lines.

Reply:
How are you only adding labels for changeStreams? I don't see any if statement for that.

Reply:
I guess it's better to remove the changes in this file at all, in the current version we can reach the test goals even without these changes.
The only reason why I left it as is that I wanted to make a decision whether the exception retryable or not based on the code and not the error type. Also, this line looks reasonable for retryable writes in the current implementation, but it will be changes in CSHARP-3001